test/test_profile_source_info.py

```diff
@@ -0,0 +1,77 @@
+from __future__ import absolute_import, division, print_function, unicode_literals
+import copy
+import unittest
+
+import torch
+import torch.jit
+from torch.utils import mkldnn as mkldnn_utils
+from common_utils import TestCase, run_tests, TemporaryFileName
+
+def num_events_populated_with_source_debug(events):
+    num_events_populated = 0
+    for e in events:
+        if e.source_debug_info != "":
+            num_events_populated += 1
+    return num_events_populated
+
+class TwoLayerNet(torch.jit.ScriptModule):
+    def __init__(self, D_in, H, D_out):
+        """
+        In the constructor we instantiate two nn.Linear modules and assign them as
+        member variables.
+        """
+        super(TwoLayerNet, self).__init__()
+        self.linear1 = torch.nn.Linear(D_in, H)
+        self.linear2 = torch.nn.Linear(H, D_out)
+
+    @torch.jit.script_method
+    def forward(self, x):
+        """
+        In the forward function we accept a Tensor of input data and we must return
+        a Tensor of output data. We can use Modules defined in the constructor as
+        well as arbitrary operators on Tensors.
+        """
+        h_relu = self.linear1(x).clamp(min=0)
+        y_pred = self.linear2(h_relu)
+        return y_pred
+
+
+class TestProfileSourceInfo(TestCase):
+    def test(self):
+        # N is batch size; D_in is input dimension;
+        # H is hidden dimension; D_out is output dimension.
+        N, D_in, H, D_out = 64, 1000, 100, 10
+
+        # Create random Tensors to hold inputs and outputs
+        x = torch.randn(N, D_in)
+        y = torch.randn(N, D_out)
+
+        # Construct our model by instantiating the class defined above
+        model = TwoLayerNet(D_in, H, D_out)
+
+        # Construct our loss function and an Optimizer. The call to model.parameters()
+        # in the SGD constructor will contain the learnable parameters of the two
+        # nn.Linear modules which are members of the model.
+        criterion = torch.nn.MSELoss(reduction='sum')
+        optimizer = torch.optim.SGD(model.parameters(), lr=1e-4)
+        with torch.autograd.profiler.profile() as prof:
+            for t in range(10):
+                # Forward pass: Compute predicted y by passing x to the model
+                y_pred = model(x)
+
+                # Compute and print loss
+                loss = criterion(y_pred, y)
+                print(t, loss.item())
+
+                # Zero gradients, perform a backward pass, and update the weights.
+                optimizer.zero_grad()
+                loss.backward()
+                optimizer.step()
+        num_events_populated = num_events_populated_with_source_debug(prof.function_events)
+        self.assertEqual(
+                num_events_populated,
+                90,
```

Comment:
this is a bad test - number of events might changes if we refactor ATen internals and organize functions differently.

Reply:
Sure. Will fix it.